### PR TITLE
refactor: typify utils

### PR DESCRIPTION
**Pull request by arshadali172**
## Problem

Partially addresses https://github.com/opengovsg/FormSG/issues/8

### CODE REVIEW

File: src/app/utils/custom-errors.ts
Comment (liangyuanruo) [inline src/app/utils/custom-errors.ts:4]: can we take the opportunity to JSDoc these errors properly?

**Review by liangyuanruo — APPROVED** [review]
to brush up on JSDocs and fix tests before merging

**Comment by karrui** [comment]
Might want to wait for emergency contact number feature to be release -- that feature adds an `ApplicationError` core error class for express. Do you want me to extract out that as its own PR so others can build upon it?

**Review by karrui — CHANGES_REQUESTED** [review]
The compilation is also currently failing due to the default export of `PERMISSION_LEVELS`. Might need to hunt down what is using it and append `.default` if the statement is using `require`

File: src/app/utils/custom-errors.ts
Comment (karrui) [inline src/app/utils/custom-errors.ts:1]: Might want to refactor them into their own module's error. For example, `ConflictError` can probably live inside `modules/submission/submission.errors.ts`, whilst `WebhookValidationError` can live inside `modules/webhooks/webhook.errors.ts`.

The PR for `ApplicationError` has also been merged in, can see #195

File: src/app/utils/custom-errors.ts
Comment (arshadali172) [reply src/app/utils/custom-errors.ts:4]: done

File: src/app/utils/custom-errors.ts
Comment (arshadali172) [reply src/app/utils/custom-errors.ts:1]: done.

**Review by karrui — APPROVED** [review]
lgtm

**Pull request merged**
